src/batch/src/task/task_execution.rs

```diff
@@ -281,44 +304,70 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
         Ok(())
     }
 
+    /// Change state and notify frontend for task status.
+    pub async fn change_state_notify(
```

Comment:
Is it possible to merge this method with `change_state`?

Reply:
Yes I think it's ok.

But this reminds me another problem: local mode `.execute` do not return a status stream, so it do not have to send the status change. However, we are reuse a lot of code for create_task and execute, so in this PR, for `.execute` we still create a channel and send stuff into it, while the receiver will never consume it. It's a cost and may be refactored later..

Reply:
LGTM